### PR TITLE
[2.7]  bpo-34395: Fix memory leaks caused by incautious usage of PyMem_Resize(). (GH-8756)

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://bugs.python.org/issue34395

### CODE REVIEW

File: Modules/_csv.c [hunk 1]
```diff
@@ -565,24 +565,21 @@ parse_save_field(ReaderObj *self)
...
+
+    unsigned field_size_new = self->field_size ? 2 * (unsigned)self->field_size : 4096;
+    if (field_size_new > INT_MAX) {
+        PyErr_NoMemory();
+        return 0;
     }
-    if (self->field == NULL) {
+    char *field_new = self->field;
+    PyMem_Resize(field_new, char, field_size_new);
+    if (field == NULL) {
```
Comment (ZackerySpytz) [inline Modules/_csv.c]: This should be `field_new`.

Comment (sir-sigurd) [reply Modules/_csv.c]: Yes, thanks.

**Comment by ZackerySpytz** [comment]
Also, variables need to be declared at the top of blocks (for C89 compliance).

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
It is possible to make the code in 2.7 simpler, but I prefer the similarity with 3.x.

**Comment by serhiy-storchaka** [comment]
On other side, the 2.7 code already is too different from the 3.x code. So I have simplified it a bit.

**Pull request merged**